### PR TITLE
Added performedCallback attribute for Redux state, and added a hook f…

### PR DESCRIPTION
**Pull request by PirosB3**
## Description

0x Instant currently does not have a "success callback" feature that allows developers to listen to. This callback can be used by application developers to create side effects of successfully filling an order.

Example:
A web application has an online store in DAI or USDC. Upon checkout, the application prompts the customer to buy some DAI using Instant (since their balance is too low). Once the checkout is complete, the web application can automatically close the Instant window and proceed with the checkout.

## Changes

In the current 0x Instant, the web-app has no control over if the rendering of 0x Instant widget resulted in a trade.
In the new 0x Instant (after this fix), the web-app can be called when a successful 0x trade happens (the event also contains the transaction hash) and take control of the application.

## Testing instructions

Tested manually in the developer page - I noticed Instant does not have tests :(

## Types of changes
- adds a new `onSuccess()` callback hook that can be passed when `render()` is called

### CODE REVIEW

File: packages/instant/src/redux/reducer.ts [hunk 1]
```diff
@@ -180,13 +182,14 @@ export const createReducer = (initialState: State) => {
                 }
                 return state;
             case ActionTypes.SetBuyOrderStateSuccess:
-                const successTxHash = action.data;
+                const successTxHash = action.data.txHash;
```
Comment (fragosti) [inline packages/instant/src/redux/reducer.ts]: Was this not correct??

Comment (PirosB3) [reply packages/instant/src/redux/reducer.ts]: it was, but I refatcored `data` to be an object now

**Comment by PirosB3** [comment]
@fragosti I'm not sure how you would implement this, compared to what I have done. Maybe we can speak about this as soon as the priority becomes current?

**Comment by PirosB3** [comment]
@fragosti this is ready to be reviewed again :)

File: packages/instant/src/containers/selected_asset_buy_order_state_buttons.ts [hunk 2]
```diff
@@ -107,7 +108,26 @@ const mapDispatchToProps = (
     },
 });
 
-export const SelectedAssetBuyOrderStateButtons: React.ComponentClass<SelectedAssetBuyOrderStateButtons> = connect(
+const mergeProps = (
+    connectedState: ConnectedState,
+    connectedDispatch: ConnectedDispatch,
+    ownProps: SelectedAssetBuyOrderStateButtons,
```
Comment (fragosti) [inline packages/instant/src/containers/selected_asset_buy_order_state_buttons.ts]: Is this the correct type for `ownProps`?

Comment (PirosB3) [reply packages/instant/src/containers/selected_asset_buy_order_state_buttons.ts]: So it looks like `SelectedAssetBuyOrderStateButtons` is being used as "props" for both `mapStateToProps` and `mapDispatchToProps`.

`SelectedAssetBuyOrderStateButtons` is defined as
```
export interface SelectedAssetBuyOrderStateButtons {}
```

Comment (PirosB3) [reply packages/instant/src/containers/selected_asset_buy_order_state_buttons.ts]: There also exists a suspicious `BuyOrderStateButtonProps` (without the `Selected` prefix) that contains a lot of properties. This interface is present in `buy_order_state_button.ts` and it's never imported/used within `selected_asset_buy_order_state_buttons.ts` so I assume it's different. Can you confirm?

Comment (fragosti) [reply packages/instant/src/containers/selected_asset_buy_order_state_buttons.ts]: Oh that makes sense. All good.

**Pull request merged**